terraform/product/main.tf

```diff
@@ -0,0 +1,121 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+data "juju_model" "hockeypuck" {
+  name = var.model
+}
+
+data "juju_model" "hockeypuck_db" {
+  name = var.db_model
+
+  provider = juju.hockeypuck_db
+}
+
+module "hockeypuck_k8s" {
+  source      = "../charm"
+  app_name    = var.hockeypuck.app_name
+  channel     = var.hockeypuck.channel
+  config      = var.hockeypuck.config
+  model       = data.juju_model.hockeypuck.name
+  constraints = var.hockeypuck.constraints
+  revision    = var.hockeypuck.revision
+  base        = var.hockeypuck.base
+  units       = var.hockeypuck.units
+}
+
+module "postgresql" {
+  source          = "git::https://github.com/canonical/postgresql-operator//terraform"
+  app_name        = var.postgresql.app_name
+  channel         = var.postgresql.channel
+  config          = var.postgresql.config
+  constraints     = var.postgresql.constraints
+  juju_model_name = data.juju_model.hockeypuck_db.name
+  revision        = var.postgresql.revision
+  base            = var.postgresql.base
+  units           = var.postgresql.units
+
+  providers = {
+    juju = juju.hockeypuck_db
+  }
+}
+
+module "nginx_ingress" {
+  source      = "./modules/nginx-ingress-integrator"
+  app_name    = var.nginx_ingress.app_name
+  channel     = var.nginx_ingress.channel
+  config      = var.nginx_ingress.config
+  constraints = var.nginx_ingress.constraints
+  model       = data.juju_model.hockeypuck.name
+  revision    = var.nginx_ingress.revision
+  base        = var.nginx_ingress.base
+  units       = var.nginx_ingress.units
+}
+
+module "traefik_k8s" {
+  source      = "./modules/traefik-k8s"
+  app_name    = var.traefik_k8s.app_name
+  channel     = var.traefik_k8s.channel
+  config      = var.traefik_k8s.config
+  constraints = var.traefik_k8s.constraints
+  model       = data.juju_model.hockeypuck.name
+  revision    = var.traefik_k8s.revision
+  base        = var.traefik_k8s.base
+  units       = var.traefik_k8s.units
+}
```

Comment:
Is there any reason to deploy `traefik-k8s` and `nginx-ingress-integrator` at the same time?